bundles/org.openhab.binding.ntp/src/main/java/org/openhab/binding/ntp/internal/handler/NtpHandler.java

```diff
@@ -54,217 +54,183 @@
-    private BigDecimal port;
-    /** refresh interval */
-    private BigDecimal refreshInterval;
-    /** NTP refresh frequency */
-    private BigDecimal refreshNtp = new BigDecimal(0);
-    /** Timezone */
-    private TimeZone timeZone;
-    /** Locale */
-    private Locale locale;
+    private ZoneId defaultTimeZoneId;
+    private ZoneId timeZoneId;
 
     /** NTP refresh counter */
     private int refreshNtpCount = 0;
     /** NTP system time delta */
     private long timeOffset;
 
-    private ChannelUID dateTimeChannelUID;
-    private ChannelUID stringChannelUID;
-
-    public NtpHandler(final Thing thing, final LocaleProvider localeProvider) {
+    public NtpHandler(final Thing thing, final TimeZoneProvider timeZoneProvider) {
         super(thing);
-        this.localeProvider = localeProvider;
+        this.defaultTimeZoneId = timeZoneProvider.getTimeZone();
+        this.timeZoneId = defaultTimeZoneId;
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        // No specific commands tied to this, but we will trigger an update
-        this.refreshNtpCount = 0;
-        refreshTimeDate();
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing channel '{}' for '{}'.", channelUID.getId(), getThing().getUID());
+            refreshTimeDate();
+        }
     }
 
     @Override
     public void initialize() {
-        try {
-            logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
+        logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
 
-            Configuration config = getThing().getConfiguration();
-            hostname = config.get(PROPERTY_NTP_SERVER_HOST).toString();
-            port = (BigDecimal) config.get(PROPERTY_NTP_SERVER_PORT);
-            refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
-            refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
-            refreshNtpCount = 0;
+        NtpThingConfiguration config = getConfigAs(NtpThingConfiguration.class);
+        configuration = config;
 
-            try {
-                Object timeZoneConfigValue = config.get(PROPERTY_TIMEZONE);
-                if (timeZoneConfigValue != null) {
-                    timeZone = TimeZone.getTimeZone(timeZoneConfigValue.toString());
-                } else {
-                    timeZone = TimeZone.getDefault();
-                    logger.debug("{} using default TZ '{}', because configuration property '{}' is null.",
-                            getThing().getUID(), timeZone, PROPERTY_TIMEZONE);
-                }
-            } catch (Exception e) {
-                timeZone = TimeZone.getDefault();
-                logger.debug("{} using default TZ '{}' due to an occurred exception: ", getThing().getUID(), timeZone,
-                        e);
-            }
+        refreshNtpCount = 0;
 
+        String timeZoneConfigValue = config.getTimeZone();
```

Comment:
If you use the defaults directly, you can access the fields here  and omit most of the null checks/function calls.

Reply:
I can't define a default for the configuration because the expected default is the timezone provided by the TimeZoneProvider.